posts/wordpress-to-jamstack.md

```diff
@@ -0,0 +1,152 @@
+
+I originally went with [Github Actions](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#schedule) but I found it to be very unreliable with running at the correct time so I was finding myself on Sundays making the calls manually until I found EasyCron. They have an integration with Vercel that made it easy to hook up with my site.
+
+# Database
+
+I wanted to learn about Cassandra and document style databases so I signed up for the new [DataStax Astra Serveless](https://www.datastax.com/blog/2021/03/astra-serverless-heres-what-you-should-know) database. This is probably overkill and could have gone with something simpler but I wanted to learn from this experience so I chose something I was not familiar with. With this approach when someone visits the website I make a call to the database and retrieve the video data from there instead of YouTube API.
+
+## NodeJS API
+
+```javascript
+const { createClient } = require("@astrajs/collections");
+
+export default async (_req, res) => {
+  // create an Astra client
+  const astraClient = await createClient({
+    astraDatabaseId: process.env.ASTRA_DB_ID,
+    astraDatabaseRegion: process.env.ASTRA_DB_REGION,
+    applicationToken: process.env.ASTRA_DB_APPLICATION_TOKEN,
+  });
+  const videoCollection = astraClient
+    .namespace(process.env.ASTRA_DB_KEYSPACE)
+    .collection(process.env.VIDEOS_COLLECTION);
+  const vids = await videoCollection.find({});
+
+  res.status(200);
+  res.send(JSON.stringify(vids));
+};
+```
+
+With the new serverless database that they offer it allows me to have a database for the Church's website and costs nothing. DataStax gives you $25 monthly credit and since the storage and read/writes are so low right now that the monthly credit covers it.
+
+# Contact Us
+
+On the old site their contact us form was no longer working and they wanted to have that feature back on their website. I ended up creating a simple form and adding a Google Captcha to it to help cutdown on potential spam they could get. The only thing I was unsure of was how to actually send the form responses. If this had been hosted on a traditional hosting platform I would have just used PHP to send the responses. I have a template for this on my [Github](https://github.com/blandfried/phpMailForm) that I put together a few years ago that might need some updates but it did the job.
+
+Since I am using serverless functions and NodeJS I had to take a different approach. I ended up signing up for an account on SendGrid because they have a very generous free tier and using [nodemailer](https://nodemailer.com/about/). Once I signed up on SendGrid and got the smtp settings it was pretty straightforward. The contact form calls the api I created and then that creates the email based on the template I defined and sends it to the Church's inbox.
```

Comment:
```suggestion
Since I am using serverless functions and NodeJS I had to take a different approach. I ended up signing up for an account on SendGrid because they have a very generous free tier and using [nodemailer](https://nodemailer.com/about/). Once I signed up on SendGrid and got the SMTP settings it was pretty straightforward. The contact form calls the API I created and then that creates the email based on the template I defined and sends it to the Church's inbox.
```

Reply:
*Node.js